packages/mason_cli/lib/src/commands/unbundle.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:convert';
+
+import 'package:args/command_runner.dart';
+import 'package:mason/mason.dart';
+import 'package:mason_cli/src/command.dart';
+import 'package:mason_cli/src/commands/bundle.dart';
+import 'package:path/path.dart' as path;
+import 'package:universal_io/io.dart';
+
+/// {@template unbundle_command}
+/// `mason unbundle` command exposes the ability to generate brick templates
+/// from bundles.
+/// {@endtemplate}
+class UnbundleCommand extends MasonCommand {
+  /// {@macro unbundle_command}
+  UnbundleCommand({Logger? logger}) : super(logger: logger) {
+    argParser
+      ..addOption(
+        'output-dir',
+        abbr: 'o',
+        help: 'Directory where to output the generated brick template.',
+        defaultsTo: '.',
+      )
+      ..addOption(
+        'type',
+        abbr: 't',
+        help: 'Type of input bundle.',
+        allowed: ['universal', 'dart'],
+        defaultsTo: 'universal',
+      );
+  }
+
+  @override
+  final String description = 'Generates a brick template from a bundle.';
+
+  @override
+  final String name = 'unbundle';
+
+  @override
+  Future<int> run() async {
+    if (results.rest.isEmpty) {
+      throw UsageException(
+        'path to the bundle must be provided',
+        usage,
+      );
+    }
+    final file = File(results.rest.first);
+    if (!file.existsSync()) {
+      throw BundleNotFoundException(file.path);
+    }
+
+    final outputDir = results['output-dir'] as String;
+    final bundleType = (results['type'] as String).toBundleType();
+
+    final bundleName = path.basenameWithoutExtension(file.path);
+    final unbundleDone = logger.progress('Unbundling $bundleName');
+
+    try {
+      late final MasonBundle bundle;
+      switch (bundleType) {
+        case BundleType.dart:
+          bundle = await _parseDartBundle(file);
+          break;
+        case BundleType.universal:
+          bundle = await _parseUniversalBundle(file);
+          break;
+      }
+      unpackBundle(bundle, Directory(outputDir));
+      unbundleDone('Unbundled ${bundle.name}');
+      logger
+        ..info(
+          '${lightGreen.wrap('✓')} '
+          'Generated 1 brick:',
+        )
+        ..detail('  ${bundle.name} ➤ ${canonicalize(outputDir)}');
+    } catch (_) {
+      unbundleDone();
+      rethrow;
+    }
+
+    return ExitCode.success.code;
+  }
+}
+
+Future<MasonBundle> _parseDartBundle(File bundleFile) async {
+  final rawBundle = await bundleFile.readAsString();
+  final bundleJson = json.decode(
```

Comment:
We should probably add a fromDartBundle to MasonBundle and perform the decode in an isolate so that for large bundles we don’t overload the main thread.

Reply:
Definitely!
 The same should apply for universal bundle serialization/de-serialization, right?

Reply:
Yup it should already be handled for the universal bundle